public/app/features/dashboard/services/PublicDashboardDataSource.ts

```diff
@@ -155,4 +159,24 @@ export class PublicDashboardDataSource extends DataSourceApi<DataQuery, DataSour
   testDatasource(): Promise<TestDataSourceResponse> {
     return Promise.resolve({ message: '', status: '' });
   }
+
+  // Try to get the browser location otherwise return blank
+  getBrowserLocation(): string {
+    // Older browser does not the internationalization API
+    if (!window.Intl) {
+      return '';
+    }
+
+    const dateFormat = window.Intl.DateTimeFormat();
+    if (!dateFormat.resolvedOptions) {
+      return '';
+    }
+
+    const options = dateFormat.resolvedOptions();
+    if (!options.timeZone) {
+      return '';
+    }
+
+    return options.timeZone;
```

Comment:
you can do the same code in this way:

```suggestion
    return window.Intl?.DateTimeFormat().resolvedOptions()?.timeZone || ''
 
```